### ISSUE TITLE
NPC Skins not changing

### ISSUE THREAD

**Issue by ScrabbleScratch** [CitizensDev/Citizens2#2450]
When I'm trying to change an NPC skin, the new skin never shows up. I've tried to do it with every form of the command, doing it with a player name to set the players skin, using the --url argument to parse the link to the skin from mineskin.org, or parsing the skin data and signature with the -t flag within the command, but none of these worked for me. I have a Paper home server, where I'm trying to make it work. First I tried to do it in my main server, but when I saw that it wasn't working i created a brand new server without any other plugin than Citizen, but it didn't work either.

Here's the log from citizen when I try to change an NPC skin: https://pastebin.com/0BQP2V92
I am using **Minecraft 1.16.4** with **paper-1.16.4-416** and **Citizens-2.0.27-b2226**

**Comment by mcmonkey4eva** [comment]
Please post the full content of `logs/latest.log`, not just that snippet. Consider also asking for help on Discord @ https://discord.gg/Q6pZGSR

**Comment by croixfish** [comment]
hi there!! i'm having a similar issue.. also, sorry, I have little to no knowledge about all of this stuff and i'm too stubborn to ask my coder bf for help so. here I am lol. i probably messed something up, but here's the full latest log! 
[latest.log](https://github.com/CitizensDev/Citizens2/files/5862967/latest.log)

**Comment by mcmonkey4eva** [comment]
- Remove SkinsRestorer, that is a plugin that exists to support illegal piracy.
- Get rid of the Bukkit jar (run a malware scan if you downloaded that from somewhere, that's generally only distributed by really shady sites... Bukkit shut down in 2014, the only jars of it are hacked versions of Spigot's internal server files) and install a proper jar of Spigot or Paper.

**Comment by ScrabbleScratch** [comment]
Actually, I don't have the plugin you mentioned (SkinsRestorer), and I'm not using Bukkit either. I have the latest release of Paper, and I'm using plugins downloaded only from the Spigot resource page. And it still doesn't work.

And as I mentioned before, I created a brand new server with nothing but the Citizens plugin. And it didn't work either.

**Comment by mcmonkey4eva** [comment]
@ScrabbleScratch that message was to croixfish.
The original message requesting a log was to you, and you still haven't posted your own log...

**Comment by ScrabbleScratch** [comment]
I'll post it as soon as possible.
I'm looking forward to a possible solution.

**Comment by ScrabbleScratch** [comment]
Hi again!
I've just tested something about the skin issue.
The thing is that I'm using TLauncher to play minecraft so, it has a function to use the TLaincher.org skin service, that I've had enabled. So I tried to connect to my server using the Mojang skin service instead of the TLauncher one, and now the skins load properly.
So, I conclude that the skins will load correctly only if you are connected to the server from a Minecraft version or launcher that is using the Mojang skin service. It is not a plugin issue but a launcher issue.
